packages/@aws-cdk/aws-stepfunctions/lib/private/json-path.ts

```diff
@@ -312,17 +312,22 @@ function pathFromToken(token: IResolvable | undefined) {
 }
 
 /**
- * Render the string in a valid JSON Path expression.
+ * Render the string or number value in a valid JSON Path expression.
  *
- * If the string is a Tokenized JSON path reference -- return the JSON path reference inside it.
- * Otherwise, single-quote it.
+ * If the value is a Tokenized JSON path reference -- return the JSON path reference inside it.
+ * If the value is a number -- convert it to string.
+ * If the value is a string -- single-quote it.
+ * Otherwise, throw errors.
  *
  * Call this function whenever you're building compound JSONPath expressions, in
  * order to avoid having tokens-in-tokens-in-tokens which become very hard to parse.
  */
-export function renderInExpression(x: string) {
-  const path = jsonPathString(x);
-  return path ?? singleQuotestring(x);
+export function renderInExpression(x: any) {
```

Comment:
Rather than throwing an error here for other types and accepting any, x should be typed to the specific valid types:

```suggestion
export function renderInExpression(x: string | number) {
```

Reply:
`renderInExpression` can receive `string[]` here, so it should at least accept `string[]`. (Note that `listAt` returns `string[]`. )
We could restrict the input type by creating another function like `renderListInExpression`, but I think it is not necessary because the function is a "private" function.

https://github.com/gkkachi/aws-cdk/blob/ab29124d26cf8a21a1399b0193122c7b8419d00d/packages/%40aws-cdk/aws-stepfunctions/lib/fields.ts#L113-L115
https://github.com/gkkachi/aws-cdk/blob/ab29124d26cf8a21a1399b0193122c7b8419d00d/packages/%40aws-cdk/aws-stepfunctions/test/fields.test.ts#L241

Reply:
If it can accept string[], the logic isn't quite right, then. You throw an error if it's not string or number.

Reply:
I wrote tests for `renderInExpression` . https://github.com/aws/aws-cdk/pull/22431/commits/d32fbf397ef27c2a8fda867a30c37366ef43a93d